plugins/woocommerce-blocks/assets/js/blocks-registry/product-collection/register-product-collection.tsx

```diff
@@ -26,17 +27,359 @@ export interface ProductCollectionConfig extends BlockVariation {
 	};
 }
 
+const isValidCollectionConfig = ( config: ProductCollectionConfig ) => {
+	// Basic checks for the top-level argument
+	if ( typeof config !== 'object' || config === null ) {
+		console.warn(
+			'Invalid arguments: You must pass an object to __experimentalRegisterProductCollection.'
+		);
```

Comment:
I see you changed all of the errors to warns. If that was based on my comment:

>…that we should not throw an error when properties that we ignore are not provided.
> - warning instead of error - collection can still be registered, it’s just property will be ignored
> - don’t bail early (no return false) - collection will still work correctly.

then we should get back to `errors` in some cases like this one. I only meant the specific case when someone provides the property that we ignore (because in those cases registering would still happen). In some cases, like this one, we should throw an error and bail early, because we cannot successfully register collection without config or if config is not an object.

Sorry for not explaining this better or highlighting it stronger.

```suggestion
		console.error(
			'Invalid arguments: You must pass an object to __experimentalRegisterProductCollection.'
		);
		return false;
```